src/lib.rs

```diff
@@ -100,6 +100,10 @@
 //!
 //!   * `extra-args: <arg 1> [... <arg n>]`, where each space separated argument will be appended,
 //!     in order, to those arguments specified as part of the `test_cmds` function.
+//!   * `stdin: <string>`, text to be passed to the command's `stdin`. If the command exits without
+//!     having consumed all of `<string>`, an error will be raised. Note, though, that operating
+//!     system file buffers can mean that the command *appears* to have consumed all of `<string>`
+//!     without it actually having done so.
```

Comment:
E.g. You have a bunch of newlines in your input string?

Reply:
The problem is even simpler than that: the OS and/or filesystem library might buffer input. So we might write to the pipe, see that the input has disappeared, and assume that it was read by the command; but the input might be in an intermediate buffer and the program can exit without ever actually having read it. I'm not aware that we can do anything about this.

Reply:
That matches my interpretation of the docstring.

Reply:
Ah I understand. Thanks!